CMakeLists.txt

```diff
@@ -727,6 +740,24 @@ elseif("${SWIFT_HOST_VARIANT_SDK}" STREQUAL "HAIKU")
   set(SWIFT_PRIMARY_VARIANT_SDK_default  "${SWIFT_HOST_VARIANT_SDK}")
   set(SWIFT_PRIMARY_VARIANT_ARCH_default "${SWIFT_HOST_VARIANT_ARCH}")
 
+elseif("${SWIFT_HOST_VARIANT_SDK}" STREQUAL "ANDROID")
+
+  set(SWIFT_HOST_VARIANT "android" CACHE STRING
+      "Deployment OS for Swift host tools (the compiler) [android]")
+
+  if("${SWIFT_SDK_ANDROID_ARCHITECTURES}" STREQUAL "")
+    set(SWIFT_SDK_ANDROID_ARCHITECTURES ${SWIFT_HOST_VARIANT_ARCH})
+  endif()
+
+  configure_sdk_unix("Android" "${SWIFT_SDK_ANDROID_ARCHITECTURES}")
+  set(SWIFT_PRIMARY_VARIANT_SDK_default  "${SWIFT_HOST_VARIANT_SDK}")
+  set(SWIFT_PRIMARY_VARIANT_ARCH_default "${SWIFT_HOST_VARIANT_ARCH}")
+
+  if("${SWIFT_ANDROID_NDK_PATH}" STREQUAL "" AND "${SWIFT_ANDROID_NATIVE_SYSROOT}" STREQUAL "")
+    message(FATAL_ERROR "Android host compiler build: you must supply either an NDK path or a native sysroot.")
+  elseif(NOT "${SWIFT_ANDROID_NDK_PATH}" STREQUAL "" AND NOT "${SWIFT_ANDROID_NATIVE_SYSROOT}" STREQUAL "")
+    message(FATAL_ERROR "Android host compiler build: you cannot supply both an NDK path and a native sysroot.")
+  endif()
```

Comment:
This logic I just added makes sure that only one or the other is set, ie you must either use the Android NDK or a native sysroot.